src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -0,0 +1,301 @@
+            profile_jsonc = commentjson.load(fileobj)
+
+        self.profiles = profile_jsonc.get("profiles", {})
+        self.defaults = profile_jsonc.get("defaults", {})
+
+        # loading secure props is done in load_profile_properties
+        # since we want to try loading secure properties only when
+        # we know that the profile has saved properties
+        # self.load_secure_props()
+
+    def get_profile(
+        self,
+        profile_name: Union[str, None] = None,
+        profile_type: Union[str, None] = None,
+    ) -> Tuple[dict, str]:
+        if self.profiles is None:
+            self.init_from_file()
+
+        if profile_name is None and profile_type is None:
+            raise ProfileNotFound(
+                "Could not find profile as both profile_name and profile_type is not set."
+            )
+
+        if profile_name is None:
+            profile_name = self.get_profilename_from_profiletype(
+                profile_type=profile_type
+            )
+
+        props: dict = self.load_profile_properties(profile_name=profile_name)
+
+        try:
+            base_profile = self.get_profilename_from_profiletype(
+                profile_type=BASE_PROFILE
+            )
+        except ProfileNotFound:
+            if self.type == TEAM_CONFIG:
+                warnings.warn(
+                    f"Base profile not found in {self.filepath}",
+                    ProfileNotFoundWarning,
+                )
+        else:
+            base_props = self.load_profile_properties(profile_name=base_profile)
+            props.update(base_props)
+
+        return props, profile_name
+
+    def autodiscover_config_dir(self):
+        """
+        Autodiscover Zowe z/OSMF Team Config files by going up the path from
+        current working directory
+
+        Return path if it finds the config directory,
+        Else, it returns None
+        """
+
+        current_dir = CURRENT_DIR
+
+        while True:
+            path = os.path.join(current_dir, self.filename)
+
+            if os.path.isfile(path):
+                self.location = current_dir
+                return
+
+            # check if have arrived at the root directory
+            if current_dir == os.path.dirname(current_dir):
+                break
+
+            current_dir = os.path.dirname(current_dir)
+
+        raise FileNotFoundError(f"Could not find the file {self.filename}")
+
+    def get_profilename_from_profiletype(self, profile_type: str) -> str:
+        """
+        Return exact profilename of the profile to load from the mentioned type
+
+        First tries to look into the defaults, if not found,
+        then it tries to iterate through the profiles
+        """
+        # try to get the profilename from defaults
+        try:
+            profilename = self.defaults[profile_type]
+        except KeyError:
+            warnings.warn(
+                "Given profile type has no default profilename",
+                ProfileParsingWarning,
```

Comment:
Should we print the given profile type in the warning ?
`Given profile type 'base' has no default profilename"`
It may be helpful 😋